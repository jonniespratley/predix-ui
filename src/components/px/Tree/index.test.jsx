import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import PxTree from './';

let data = [{
  id: 1,
  label: "All Categories",
  children: [
    {
      id: '2',
      label: "For Sale",
      children: [
        {
          id: '2-1',
          label: "Category 1"
        },
        {
          id: '2-2',
          label: "Category 2"
        },
        {
          id: '2-3',
          label: "Category 3"
        }
      ]
    }
  ]
}];



describe('Tree', () => {
  test('should render', () => {
    const wrapper = shallow(<PxTree items={data}/>);
    console.log(wrapper.debug());
  //  expect(true).toEqual(true);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <PxTree items={data}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
