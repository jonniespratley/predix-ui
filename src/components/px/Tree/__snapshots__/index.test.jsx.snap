// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree matches snapshot 1`] = `
.c3 {
  color: var(--iron-icon-stroke-color,inherit);
  fill: var(--iron-icon-fill-color,none);
  stroke: var(--iron-icon-stroke-color,currentColor);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  vertical-align: middle;
  width: var(--iron-icon-width);
  height: var(--iron-icon-height);
}

.c3 > svg {
  color: var(--iron-icon-stroke-color,inherit);
  fill: var(--iron-icon-fill-color,none);
  stroke: var(--iron-icon-stroke-color,currentColor);
  position: relative;
  vertical-align: middle;
}

.c3.hidden {
  display: none;
}

.c0 {
  list-style: none;
  padding: 0;
  margin: 0;
  line-height: 1.8rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 1rem;
  height: 1rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

.c4 {
  margin: 0;
  list-style: none;
  padding-left: 1.5rem !important;
  display: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding-left: 1.75rem;
}

.c1:hover {
  color: var(--px-tree-text-color--hover,black);
  background-color: var(--px-tree-background-color--hover,lightgray);
}

.c1:hover {
  color: var(--px-tree-text-color--hover,#000);
  background-color: var(--px-tree-background-color--hover,#d3d3d3);
}

<ul
  className="px-tree"
  style={
    Object {
      "margin": 0,
      "padding": 0,
    }
  }
>
  <li
    className="c0"
    data-branch={true}
    data-leaf={false}
    data-open={false}
    onClick={[Function]}
    open={false}
    selected={false}
  >
    <div
      className="c1"
      data-id={1}
      selected={false}
    >
      <span
        className="c2"
      >
        <i
          className="px-icon px-utl-chevron-right c3"
          color={null}
          style={
            Object {
              "height": 16,
              "width": 16,
            }
          }
        >
          <svg
            dangerouslySetInnerHTML={
              Object {
                "__html": "<g id=\\"px-utl-chevron-right\\"><path stroke-linejoin=\\"round\\" d=\\"M6.2 13.2l5.4-5.5-5.5-5.5\\"/></g>",
              }
            }
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "display": "block",
                "height": 16,
                "pointerEvents": "none",
                "width": 16,
              }
            }
            viewBox="0 0 16 16"
          />
        </i>
      </span>
      <span
        onClick={[Function]}
      >
        All Categories
      </span>
       
    </div>
    <ul
      className="c4"
      open={false}
    />
  </li>
</ul>
`;
